lms/djangoapps/courseware/views/views.py

```diff
@@ -726,6 +726,12 @@ def _progress(request, course_key, student_id):
 
     # checking certificate generation configuration
     enrollment_mode, is_active = CourseEnrollment.enrollment_mode_for_user(student, course_key)
+
+    # We need to have id verified to show generate cert button
+    # in the case where courseMode is of Verified variety.
+    is_id_verified = SoftwareSecurePhotoVerification.user_is_verified(student) or \
+        enrollment_mode and not CourseMode.has_verified_mode(enrollment_mode)
```

Comment:
I'm finding this difficult to parse. Either the student must have a valid photo verification, or the course mode must not be verified? How is `is_id_verified` used later down the line? The name of the variable is confusing given the logic used to determine its value.